### DISCUSSION TITLE
Change CURLMOPT_MAX_TOTAL_CONNECTIONS and CURLMOPT_MAX_HOST_CONNECTIONS options

### DISCUSSION THREAD

**Discussion by pkropachev** [curl/curl#13158] [Q&A] [answered]
Hello!

In documentation for options [CURLMOPT_MAX_HOST_CONNECTIONS](https://curl.se/libcurl/c/CURLMOPT_MAX_HOST_CONNECTIONS.html) and [CURLMOPT_MAX_TOTAL_CONNECTIONS](https://curl.se/libcurl/c/CURLMOPT_MAX_TOTAL_CONNECTIONS.html) nothing say about ability to change these options while the life of curl multi handle (`CURLM`).
Could you please clarify if it's possible to change these options during the life of `CURLM`? If so, how will new values be applied? For example initial value for `CURLMOPT_MAX_HOST_CONNECTIONS` was `10`, then it's changed to `5`. Will curl manage it by itself and narrow connections down?

Thanks!

**Comment by jay** [comment]
The code is not written for that. CURLMOPT_MAX_HOST_CONNECTIONS here is the logic:

https://github.com/curl/curl/blob/5ce164e0e9290c96eb7d502173426c0a135ec008/lib/url.c#L3689-L3717

**Reply by pkropachev** [reply]
Thanks @jay!
Let's currently consider it like changes these options won't have any effect. I'll check with test application.

**Reply by bagder** [reply]
I would rather say that changes *will* have an effect. The next time the value is checked, it takes effect.

**Reply by pkropachev** [reply]
Thanks @bagder!
But for me still not clear how will curl decrease number of simultaneously connection, e.g. in case change it from `10` to `5`?

**Reply by bagder** [reply]
It does nothing initially. The check is only done when deciding if a new connection can be setup, so if you already have 10 connections when you lower the max to 5, nothing will happen to your already setup connections. But if three of them finish and you want to create a new transfer, it cannot create a new connection anymore since it is now already maxed out.

**Comment by bagder** [comment, marked as answer]
Here is my attempt at clarifying this: https://github.com/curl/curl/pull/13176

**Reply by pkropachev** [reply]
Thanks Daniel!
